shared/js/moz_intl.js

```diff
@@ -71,71 +71,50 @@ global.mozIntl = {
   DateTimeFormat: function(locales, options) {
     const resolvedOptions = Object.assign({}, options);
 
-    if (resolvedOptions.dayperiod) {
-      if (resolvedOptions.hour === undefined) {
-        resolvedOptions.hour = 'numeric';
-      }
-    }
-
     if (resolvedOptions.hour !== undefined &&
         resolvedOptions.hour12 === undefined) {
       resolvedOptions.hour12 = navigator.mozHour12;
     }
 
-    if (resolvedOptions.dayperiod === undefined &&
-        resolvedOptions.hour12 === true) {
-      resolvedOptions.dayperiod = true;
-    }
 
     var intlFormat = new Intl.DateTimeFormat(locales, resolvedOptions);
 
     resolvedOptions.locale = intlFormat.resolvedOptions().locale;
     resolvedOptions.hour12 = intlFormat.resolvedOptions().hour12;
```

Comment:
Why are you overwriting `hour12` here unconditionally?  Should this code check if line 76 was run first?